File: api-openbis-java/source/java/ch/ethz/sis/openbis/generic/asapi/v3/typegroup/create/TypeGroupCreation.java
/*
 *  Copyright ETH 2025 Zürich, Scientific IT Services
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 */

package ch.ethz.sis.openbis.generic.asapi.v3.typegroup.create;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.ObjectToString;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.create.ICreation;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.create.IObjectCreation;
import ch.systemsx.cisd.base.annotation.JsonObject;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.Map;

@JsonObject("as.dto.typegroup.create.TypeGroupCreation")
public class TypeGroupCreation implements ICreation, IObjectCreation
{
    private static final long serialVersionUID = 1L;

    @JsonProperty
    private String name;

    @JsonProperty
    private boolean managedInternally;

    @JsonProperty
    private Map<String, String> metaData;

    @JsonIgnore
    public String getName()
    {
        return name;
    }

    public void setName(String name)
    {
        this.name = name;
    }

    @JsonIgnore
    public boolean isManagedInternally()
    {
        return managedInternally;
    }

    public void setManagedInternally(boolean managedInternally)
    {
        this.managedInternally = managedInternally;
    }

    @JsonIgnore
    public Map<String, String> getMetaData()
    {
        return metaData;
    }

    public void setMetaData(Map<String, String> metaData)
    {
        this.metaData = metaData;
    }

    @Override
    public String toString()
    {
        return new ObjectToString(this).append("name", name)
                .append("managedInternally", managedInternally).toString();
    }
}

File: api-openbis-javascript/src/v3/as/dto/typegroup/create/TypeGroupCreation.js
define([ "stjs", "util/Exceptions" ], function(stjs, exceptions) {
	var TypeGroupCreation = function() {
	};
	stjs.extend(TypeGroupCreation, null, [], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.typegroup.create.TypeGroupCreation';
		constructor.serialVersionUID = 1;
		prototype.code = null;
		prototype.metaData = null;
        prototype.managedInternally = null;

		prototype.getCode = function() {
			return this.code;
		};
		prototype.setCode = function(code) {
			this.code = code;
		};
        prototype.getMetaData = function() {
            return this.metaData;
        };
        prototype.setMetaData = function(metaData) {
            this.metaData = metaData;
        };
        prototype.isManagedInternally = function() {
            return this.managedInternally;
        };
        prototype.setManagedInternally = function(managedInternally) {
            this.managedInternally = managedInternally;
        };
        prototype.toString = function () {
            return '[' + this.code + ', ' + this.managedInternally + ']';
        }
	}, {
		metaData: {
            name: "Map",
            arguments: ["String", "String"]
        }
	});
	return TypeGroupCreation;
})